kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -359,53 +386,148 @@ protected ProcurementCardDefault retrieveProcurementCardDefault(String creditCar
      */
     @Override
     public boolean autoApproveProcurementCardDocuments() {
-        //if auto approve is turned on
-        boolean autoApproveOn = super.parameterService.getParameterValueAsBoolean(
-                ProcurementCardAutoApproveDocumentsStep.class,
-                ProcurementCardDocumentRuleConstants.AUTO_APPROVE_DOCUMENTS_IND);
+        // if auto approve is turned on
+        boolean autoApproveOn = parameterService.getParameterValueAsBoolean(ProcurementCardAutoApproveDocumentsStep.class, ProcurementCardDocumentRuleConstants.AUTO_APPROVE_DOCUMENTS_IND);
 
-        if (!autoApproveOn) { // no auto approve?  then skip out of here...
+        if (!autoApproveOn) { // no auto approve? then skip out of here...
             return true;
         }
 
+        StringBuffer autoApproveErrorStr = new StringBuffer();
+
+        // get Parameter information
+        Timestamp currentDate = dateTimeService.getCurrentTimestamp();
+        int autoApproveNumberDays = Integer.parseInt(parameterService.getParameterValueAsString(ProcurementCardAutoApproveDocumentsStep.class, ProcurementCardDocumentRuleConstants.AUTO_APPROVE_NUMBER_OF_DAYS));
```

Comment:
Ideally this method would sit on the ParameterService in rice, but since we're not looking to change rice code for now, it can wait.
Still the  job will stop at a standstill with a NPE if the parameter is not set or if it has a non-numeric value. (... same behavior as in KFS3 )